wsd/Storage.cpp

```diff
@@ -872,11 +881,40 @@ WopiStorage::WOPIFileInfo::WOPIFileInfo(const FileInfo &fileInfo,
     JsonUtil::findJSONValue(object, "BreadcrumbDocName", _breadcrumbDocName);
     JsonUtil::findJSONValue(object, "FileUrl", _fileUrl);
 
-    bool booleanFlag = false;
-    JsonUtil::findJSONValue(object, "IsUserLocked", booleanFlag);
-    CommandControl::LockManager::setLockedUser(booleanFlag);
+#ifdef ENABLE_FEATURE_LOCK
+    bool isUserLocked = false;
+    JsonUtil::findJSONValue(object, "IsUserLocked", isUserLocked);
+
+    if (config::getBool("feature_lock.locked_hosts[@allow]", false))
+    {
+        bool isReadOnly = false;
+        bool isCommandDisabled = false;
+        isUserLocked = false;
 
-    booleanFlag = false;
+        const std::string host = uriObject.getHost();
+        std::cerr << "host: " << host << "\n";
+        if (CommandControl::LockManager::hostExist(host))
+        {
+            isReadOnly = CommandControl::LockManager::isHostReadOnly(host);
+            isCommandDisabled = CommandControl::LockManager::isHostCommandDisabled(host);
+        }
+        else
+        {
+            LOG_INF("Could not find matching locked host so applying fallback settings");
+            isReadOnly = config::getBool("feature_lock.locked_hosts.fallback[@read_only]", false);
+            isCommandDisabled = config::getBool("feature_lock.locked_hosts.fallback[@disabled_commands]", false);
+        }
+
+        if (isCommandDisabled || isReadOnly)
+        {
+            isUserLocked = true;
+        }
+        CommandControl::LockManager::setHostReadOnly(isReadOnly);
+    }
+    CommandControl::LockManager::setLockedUser(isUserLocked);
+#endif
```

Comment:
Would be good to have an #else branch that only does a `(void) uriObject;` to avoid `error: unused parameter ‘uriObject’ [-Werror=unused-parameter]`.